trace-reader/src/main.rs

```diff
@@ -0,0 +1,84 @@
+use anyhow::Result;
+use clap::Parser;
+use rand::{seq::IteratorRandom, thread_rng};
+use rdkafka::producer::FutureProducer;
+
+mod loader;
+mod processing;
+use loader::load_trace_file;
+use processing::dispatch_trace_file;
+
+#[derive(Debug, Parser)]
+#[clap(author, version, about)]
+struct Cli {
+    #[clap(short, long)]
+    broker: String,
+
+    #[clap(short, long)]
+    username: Option<String>,
+
+    #[clap(short, long)]
+    password: Option<String>,
+
+    #[clap(short, long = "group", default_value = "trace-producer")]
+    consumer_group: String,
+
+    #[clap(short, long, default_value = "Traces")]
+    trace_topic: String,
+
+    #[clap(short, long)]
+    file_name: Option<String>,
```

Comment:
You should be able to use a `std::path::PathBuf` here.

Reply:
Fixed